message/rpc_protocol_impl.hh

```diff
@@ -177,32 +177,44 @@ auto send_message_cancellable(messaging_service* ms, messaging_verb verb, msg_ad
     if (ms->is_shutting_down()) {
         return futurator::make_exception_future(rpc::closed_error());
     }
+  try {
     auto rpc_client_ptr = ms->get_rpc_client(verb, id);
     auto& rpc_client = *rpc_client_ptr;
 
-    auto c = std::make_unique<seastar::rpc::cancellable>();
-    auto& c_ref = *c;
-    auto sub = as.subscribe([c = std::move(c)] () noexcept {
-        c->cancel();
-    });
-    if (!sub) {
-        return futurator::make_exception_future(abort_requested_exception{});
-    }
+    struct state {
+        seastar::rpc::cancellable c;
+        std::exception_ptr ex;
+        optimized_optional<abort_source::subscription> sub;
+
+        explicit state(abort_source& as) {
+            as.check();
+            sub = as.subscribe([this] (const std::optional<std::exception_ptr>& opt_ex) noexcept {
+                c.cancel();
```

Comment:
This adds an allocation. Do we have data-plane callers using this?

The data plane should likely manage its about sources on its own. This is convenient for the control plane.

Reply:
No, it's called only from `send_gossip_echo` and `send_schema_check`.

Reply:
Where is the extra allocation by the way?
We currently have `std::make_unique<seastar::rpc::cancellable>()`
and this changes it `std::make_unique<state>(as)`